File: javascript/02-remove_Array_Value.js
const removeElement = function (nums, val) {
  let i = 0;
  for (let j = 0; j < nums.length; j++) {
    if (nums[j] !== val) {
      nums[i++] = nums[j];
    }
  }
  return i;
};

const removeElement2 = (nums, val) => {
  let i = 0;
  for (let j = 0; j < nums.length; j++) {
    if (nums[j] !== val) {
      nums[i] = nums[j];
      i++;
    }
  }
  return i;
};

const removeElement3 = (nums, val) => {
  let i = 0;
  let n = nums.length;
  while (i < n) {
    if (nums[i] === val) {
      nums[i] = nums[n - 1];
      n--;
    } else {
      i++;
    }
  }
  return n;
};

nums = [3, 2, 2, 3];
val = 3;
console.log(removeElement3(nums, val));
console.log(nums);

File: python/02-remove_Array_Value.py
def removeElement(nums, val):
    i = 0
    for j, num in enumerate(nums):
        if (num != val):
            nums[i] = nums[j]
            i += 1
    return i

# 2 approach


def removeElement2(nums, val):
    i = 0
    n = len(nums)

    while i < n:
        if (nums[i] == val):
            nums[i] = nums[n-1]
            n -= 1
        else:
            i += 1
    return n


nums = [3, 2, 2, 3]
val = 3
print(removeElement2(nums, val))
print(nums)
print("hello")
